### PR TITLE
refactor(connectivity_plus)!: two-package federated architecture

### PR DESCRIPTION
**Pull request by miquelbeltran**
## Description

Refactor work for `connectivity_plus`.

I discovered some tests in the web implementation that were not used/maintained, so I deleted them.

## Related Issues

- Related #1226 

## Checklist

- [x] I read the Contributor Guide and followed the process outlined there for submitting PRs.
- [x] I titled the PR using [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0).
- [ ] I did not modify the `CHANGELOG.md` nor the `pubspec.yaml` files.
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [x] Yes, this is a breaking change (please indicate that with a `!` in the title as explained in [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0)).
- [ ] No, this is *not* a breaking change.

### CODE REVIEW

**Comment by miquelbeltran** [comment]
can't figure out why the Windows build fails, will boot up my Windows PC and check there :joy:

**Comment by miquelbeltran** [comment]
> Looks good to me. Let's just fix the import in tests as advised by the analyzer action.

Seem unrelated to this package, will address on another PR

File: packages/connectivity_plus/connectivity_plus/lib/connectivity_plus.dart [hunk 1]
```diff
@@ -10,6 +10,8 @@ import 'package:connectivity_plus_platform_interface/connectivity_plus_platform_
 export 'package:connectivity_plus_platform_interface/connectivity_plus_platform_interface.dart'
     show ConnectivityResult;
 
+export 'src/connectivity_plus_linux.dart';
+
```
Comment (heeyunlee) [inline packages/connectivity_plus/connectivity_plus/lib/connectivity_plus.dart:14]: The newest version of the connectivity_plus package does not work after the 3.0.0 version release. Shouldn't you have `export 'src/connectvity_plus_web.dart'` here?

Comment (miquelbeltran) [reply packages/connectivity_plus/connectivity_plus/lib/connectivity_plus.dart:14]: See https://github.com/fluttercommunity/plus_plugins/issues/1270

Comment (miquelbeltran) [reply packages/connectivity_plus/connectivity_plus/lib/connectivity_plus.dart:14]: But yes, probably the export web is missing here. Please submit a PR or an issue if you can!

Comment (nohli) [reply packages/connectivity_plus/connectivity_plus/lib/connectivity_plus.dart:14]: https://github.com/fluttercommunity/plus_plugins/pull/1278

**Pull request merged**